### PR TITLE
Go add registry

### PR DESCRIPTION
**Pull request by Anghul**

### CODE REVIEW

File: be1-go/channel/register/mod.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package register
```
Comment (nkcr) [inline be1-go/channel/register/mod.go:1]: `registry` would be better, what do you think ?

File: be1-go/channel/register/mod.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+//   }
+//
+//   // register the "elect" type
+//   registry.register("elect", execElect, messagedata.ConsensusElect{})
+//
+//   // when we need to process a message we call "processMsg"
+//   err = registry.processMsg(msg)
+//
+func (m MessageRegistry) Register(action string, f func(message.Message, interface{}) error, c interface{}) {
+	m.Registry[action] = CallbackData{
```
Comment (nkcr) [inline be1-go/channel/register/mod.go:39]: In fact we could simplify it and take the "action" from `c` directly. You do that by having a map of `reflect.Type`:

```go
map[reflect.Type]
```

Then you can use:

```go
key := reflect.TypeOf(c)
map[key] = CallbackData{...}
```

Comment (nkcr) [reply be1-go/channel/register/mod.go:39]: And add a function:

```go
func (m MessageRegistry) Process(c interface{}) error {
  key := reflect.TypeOf(c)

  data, found := m.Registry[key]
  if !found {...}

  concreteType := reflect.New(reflect.ValueOf(data.ConcreteType).Type()).Interface()

	err := msg.UnmarshalData(&concreteType)
	if err != nil {
		return xerrors.Errorf("failed to unmarshal data: %v", err)
	}

	err = data.Callback(c, concreteType)
	if err != nil {
		return xerrors.Errorf("failed to call message '%s': %v", key, err)
	}

  return nil
}

Comment (nkcr) [reply be1-go/channel/register/mod.go:39]: Then the attribute of `MessageRegistry` (`Registry`) can be private.

File: be1-go/channel/consensus/mod.go [hunk 3]
```diff
@@ -103,6 +107,25 @@ func NewChannel(channelID string, hub channel.HubFunctionalities, log zerolog.Lo
 		consensusInstances: make(map[string]*ConsensusInstance),
 		messageStates:      make(map[string]*MessageState),
 	}
+
+	newChannel.register = newChannel.NewConsensusRegistry()
```
[the remark this answers is not in the document]
Comment (Anghul) [reply be1-go/channel/consensus/mod.go:111]: We need the channel for the registry, as all of the functions processing the messages are functions of the channel, and it therefore need to be created before the registry.

**Review by nkcr — DISMISSED** [review]
LGTM 🚀

**Review by nkcr — APPROVED** [review]

**Pull request merged**